frontend/src/metabase/components/FieldValuesWidget/utils.ts

```diff
@@ -247,22 +232,10 @@ export function getTokenFieldPlaceholder({
   options: FieldValue[];
   valuesMode: ValuesMode;
 }) {
-  if (placeholder) {
-    return placeholder;
+  if (shouldList({ parameter, fields, disableSearch }) && !_.isEmpty(options)) {
```

Comment:
I'm not sure we'd still need this if check anymore. On paper `if (placeholder)` is already replaced by [this change](https://github.com/metabase/metabase/pull/34910/files#diff-b1cf4299750fb16b84cd22730312df1238fd6816eb26bd8d03b44a3aa5da337dR437-R438).

And `shouldList({ parameter, fields, disableSearch }) && !_.isEmpty(options)` is basically `hasList(...)` while this function is only called in the [else block](https://github.com/metabase/metabase/pull/34910/files#diff-b1cf4299750fb16b84cd22730312df1238fd6816eb26bd8d03b44a3aa5da337dR431) which should be something along the line `!hasList(...)` if I read the logic correctly since the first 2 conditions are alreaady `hasList(...)`

Can you explain your reasoning here? I think it makes more sense to remove this block altogether.

Another observation is that `SEARCH_THE_LIST_PLACEHOLDER` is also used in `frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.tsx`. I suspect that is already enough to not have this block. And we might also want to declare `SEARCH_THE_LIST_PLACEHOLDER` in `FieldValuesWidget` instead.

Reply:
@WiNloSt I think you're confused by github placing these lines next to each other in the diff viewer. I _have_ removed the `if (placeholder) { return placeholder; } ` block.

This line, 235, replaces [these lines](https://github.com/metabase/metabase/pull/34910/files#diff-c7d2638c5a506ff4d59c4d8d0f346c43cc1e46cafb689f9a7d0d6fb555d967baL257-L263) (257-263).

btw your link to the [else block](https://github.com/metabase/metabase/pull/34910/files#diff-b1cf4299750fb16b84cd22730312df1238fd6816eb26bd8d03b44a3aa5da337dR431) doesn't navigate to any code in my browser.

Reply:
@calherries 
Seems linking to collapsed code doesn't work. Here's the mentioned else block.
![image](https://github.com/metabase/metabase/assets/1937582/31b32306-b703-43f7-87b3-75f0a82e642e)

> Another observation is that `SEARCH_THE_LIST_PLACEHOLDER` is also used in `frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.tsx`. I suspect that is already enough to not have this block. And we might also want to declare `SEARCH_THE_LIST_PLACEHOLDER` in `FieldValuesWidget` instead.

This one is more relevant than the `if (placeholder)` thing.

The bottom line is that could you double-check if we still need to have `SEARCH_THE_LIST_PLACEHOLDER` inside `utils.ts` as I think it's impossible for `shouldList({ parameter, fields, disableSearch }) && !_.isEmpty(options)` to be true inside `getTokenFieldPlaceholder` based on how `getTokenFieldPlaceholder` get called (in the mentioned else block)

Reply:
Yep, double checking. 

```
We want to prove that:
"TokenWidget is called"
=> line 235 condition is false.

Or the contrapositive:
line 235 condition is true
=> "TokenWidget is not called"

We can substitute
line 235 condition is true <=> shouldList(...) && !_.isEmpty(options)
"TokenWidget is not called" <=> isListMode && !_.isEmpty(options)

So we need to prove
shouldList(...) && !_.isEmpty(options)
=> isListMode && !_.isEmpty(options)

this is not the case if `shouldList(...) && isSearchable(...)`

Simplifying:
shouldList(...) && !_.isEmpty(options)
=> isListMode

To disprove this we can find one instance where 
shouldList(...) && !_.isEmpty && !isListMode

Let's consider
isListMode  = !disableList && valuesMode === "list" && !forceTokenField

so when disableList or forceTokenField is true then it's possible for isListMode to be false but shouldList to be true.

therefore
forceTokenField && shouldList(...) => shouldList(...) && !_.isEmpty(options) && !isListMode

which contradicts the contrapositive
line 235 condition is true
=> "TokenWidget is not called"

I found an instance of this:
https://www.loom.com/share/06da2881c03d4a488d97e5967bdbbe8f?sid=8b8600ac-fb72-437d-97fc-b89108b1ef68

Reply:
Awesome investigation. Thanks for putting this together. I agree that it's necessary. I just thought there might be a simpler way to deal with not having to share `SEARCH_THE_LIST_PLACEHOLDER` to another file, but this looks like the best option at the moment.